LINK-93, LINK-94, LINK-95 | Event keywords

jorilindell opened the pull request:
## Description
Add GraphQl queries to get events
Generic CheckboxGroupField component
KeywordSelector component to search and select keywords
Audience and classification event form sections

## Closes
[LINK-93](https://helsinkisolutionoffice.atlassian.net/browse/LINK-93)
[LINK-94](https://helsinkisolutionoffice.atlassian.net/browse/LINK-94)
[LINK-95](https://helsinkisolutionoffice.atlassian.net/browse/LINK-95)

## Screenshots
<img width="916" alt="Screenshot 2020-11-20 at 14 18 08" src="https://user-images.githubusercontent.com/24706814/99799408-4542c280-2b3b-11eb-9415-66b327ed824a.png">


<img width="901" alt="Screenshot 2020-11-20 at 14 18 16" src="https://user-images.githubusercontent.com/24706814/99799416-4a077680-2b3b-11eb-8d66-cbc16b7161ea.png">

Sartonon commented:
<img width="2080" alt="Screenshot 2020-11-22 at 16 58 31" src="https://user-images.githubusercontent.com/15219142/99907184-013ef180-2ce4-11eb-994c-24db0849af9e.png">

Seems like safari doesn't like the way form steps are handled in the dom, form gets too wide when navitagingto next step

jorilindell commented:
> <img alt="Screenshot 2020-11-22 at 16 58 31" width="2000" src="https://user-images.githubusercontent.com/15219142/99907184-013ef180-2ce4-11eb-994c-24db0849af9e.png">
> 
> Seems like safari doesn't like the way form steps are handled in the dom, form gets too wide when navitagingto next step

Fixed. Didn't happen with my 13 inch Macbook, but only with larger screens. Also fixed css issues with loading spinner and Safari

Sartonon reviewed (commented):
hmm is there a reason not to use events-helsinki-api-proxy? Altough apollo-link-rest seems handy :) Otherwise looking good!

jorilindell commented on src/common/components/keywordSelector/KeywordSelector.tsx:
```diff
@@ -0,0 +1,158 @@
+          // Make sure all selected keywords are fetched to cache
+          <KeywordQuery key={index} id={parseIdFromAtId(keyword) as string} />
+        );
+      })}
```
It's not. I think that I changed the logic and forgot to remove this

jorilindell commented on src/common/components/keywordSelector/KeywordSelector.tsx:
```diff
@@ -0,0 +1,158 @@
+  useKeywordQuery({
+    variables: {
+      id,
+      createPath: isTestEnv
```
Added getPathBuilder helper function

jorilindell commented on src/common/components/keywordSelector/KeywordSelector.tsx:
```diff
@@ -0,0 +1,158 @@
+  const { data: keywordsData } = useKeywordsQuery({
+    variables: {
+      freeText: search,
+      createPath: isTestEnv
```
Added getPathBuilder helper function

jorilindell commented on src/domain/event/formSections/audienceSection/AudienceSection.tsx:
```diff
@@ -0,0 +1,69 @@
+
+  const { data } = useKeywordSetQuery({
+    variables: {
+      createPath: isTestEnv
```
Added getPathBuilder helper function

jorilindell commented on src/domain/event/formSections/classificationSection/ClassificationSection.tsx:
```diff
@@ -0,0 +1,102 @@
+
+  const { data } = useKeywordSetQuery({
+    variables: {
+      createPath: isTestEnv
```
Added getPathBuilder helper function

jorilindell commented on src/domain/keyword/utils.ts:
```diff
@@ -0,0 +1,73 @@
+      query: KeywordDocument,
+      variables: {
+        id,
+        createPath: isTestEnv
```
Added getPathBuilder helper function

jorilindell commented:
> hmm is there a reason not to use events-helsinki-api-proxy? Altough apollo-link-rest seems handy :) Otherwise looking good!

So far events-helsinki-api-proxy is only for fetching data from the LinkedEvents. On next sprint we're going to add authentication and saving data features to the UI which events-helsinki-api-proxy doesn't support. At some there will be GraphQl API and we have to migrate to that anyway. I think that events-helsinki-api-proxy could start using that when it's available.

Sartonon commented:
> > hmm is there a reason not to use events-helsinki-api-proxy? Altough apollo-link-rest seems handy :) Otherwise looking good!
> 
> So far events-helsinki-api-proxy is only for fetching data from the LinkedEvents. On next sprint we're going to add authentication and saving data features to the UI which events-helsinki-api-proxy doesn't support. At some there will be GraphQl API and we have to migrate to that anyway. I think that events-helsinki-api-proxy could start using that when it's available.

OK!
